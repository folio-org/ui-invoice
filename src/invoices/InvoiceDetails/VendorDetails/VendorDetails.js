import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { get } from 'lodash';

import { stripesConnect } from '@folio/stripes/core';
import {
  Col,
  KeyValue,
  Row,
} from '@folio/stripes/components';

import { vendorItem } from '../../../common/resources';

const VendorDetails = ({ vendorInvoiceNo, accountingCode, resources }) => {
  const vendorName = get(resources, 'vendor.records.0.name');

  return (
    <Row>
      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-invoice.invoice.details.vendor.vendorInvoiceNo" />}
          value={vendorInvoiceNo}
        />
      </Col>

      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-invoice.invoice.details.vendor.name" />}
          value={vendorName}
        />
      </Col>

      <Col xs={3}>
        <KeyValue
          label={<FormattedMessage id="ui-invoice.invoice.details.vendor.accountingCode" />}
          value={accountingCode}
        />
      </Col>
    </Row>
  );
};

VendorDetails.manifest = {
  vendor: vendorItem,
};

VendorDetails.propTypes = {
  vendorInvoiceNo: PropTypes.string,
  accountingCode: PropTypes.string,
  // vendorId prop is used in manifest
  // eslint-disable-next-line react/no-unused-prop-types
  vendorId: PropTypes.string,
  resources: PropTypes.object.isRequired,
};

export default stripesConnect(VendorDetails);
