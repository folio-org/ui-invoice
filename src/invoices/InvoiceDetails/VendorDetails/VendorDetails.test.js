import React from 'react';
import { render } from '@testing-library/react';

import { invoice } from '../../../../test/jest/fixtures';

import VendorDetails from './VendorDetails';

jest.mock('@folio/stripes-acq-components', () => ({
  OrganizationValue: jest.fn(({ id }) => id),
}));

const { accountingCode, vendorId, vendorInvoiceNo } = invoice;
const defaultProps = {
  accountingCode,
  vendorId,
  vendorInvoiceNo,
};
const renderVendorDetails = (props = defaultProps) => render(
  <VendorDetails {...props} />,
);

describe('VendorDetails', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure for invoice vendor details', () => {
    const { asFragment } = renderVendorDetails();

    expect(asFragment()).toMatchSnapshot();
  });
});
