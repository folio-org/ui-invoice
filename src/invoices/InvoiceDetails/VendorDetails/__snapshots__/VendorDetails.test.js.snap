// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorDetails should render correct structure for invoice vendor details 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-xs-3"
    >
      <div
        class="kvRoot"
      >
        <div
          class="kvLabel"
        >
          ui-invoice.invoice.details.vendor.vendorInvoiceNo
        </div>
        <div
          class="kvValue"
          data-test-kv-value="true"
        >
          edi
        </div>
      </div>
    </div>
    <div
      class="col-xs-3"
    >
      11fb627a-cdf1-11e8-a8d5-f2801f1b9fd1
    </div>
    <div
      class="col-xs-3"
    >
      <div
        class="kvRoot"
      >
        <div
          class="kvLabel"
        >
          ui-invoice.invoice.accountingCode
        </div>
        <div
          class="kvValue"
          data-test-kv-value="true"
        >
          G64758-74834
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
